Add automated pep8 format check

kursatyurt opened the pull request:
ASTE repo was lack of an format-checker for python scripts.

This PR provides a GitHub workflow for format checking the python scripts using pep8 and following rules.

`autopep8 --recursive --in-place --aggressive --aggressive --exit-code --ignore E402 --max-line-length 120`

The rules are adopted from preCICE python binding repo (see https://github.com/precice/python-bindings/pull/82).

The workflow checks the format if it fits to pep8 standart

DavidSCN commented on .github/workflows/pep8.yml:
```diff
@@ -0,0 +1,36 @@
+        id: autopep8
+        uses: peter-evans/autopep8@v1
+        with:
+          args: --recursive --in-place --aggressive --aggressive --exit-code --ignore E402 --max-line-length 120 .
```
Do we really need the `--aggressive` here twice?

kursatyurt replied:
It gives aggressiveness level 2 between [0-4]. Allows autpep8 to do some changes to solve errors in linting.

```
Use multiple --aggressive to increase the aggressiveness level. For example, E712 requires aggressiveness level 2 (since x == True could be changed to either x or x is True, but autopep8 chooses the former).
```

DavidSCN commented:
I'm not a big fan of bots touching my code, but we can give it a trial. AFAIK the more convenient way of doing such things would be to add a formatting scripts which can be executed by the user. However, we can give it a trial, I will wait for feedback of others here. We should anyway not merge this PR before we merged larger open PRs due to merge conflicts.

kursatyurt commented:
The action bot is removed. ATM even after formatting on local ["fedora35","macOS"] the workflow fails.

My anyone tests it what is the issue?

DavidSCN commented:
The action should print out the ill formatted lines so that one gets informed about the failing file. Maybe something like [this](https://github.com/precice/dealii-adapter/blob/005933d6b45f885a64aee7ce597a3d7d81d54aff/.github/workflows/indent.yml#L22-L24) works here as well.

kursatyurt commented:
> The action should print out the ill formatted lines so that one gets informed about the failing file. Maybe something like [this](https://github.com/precice/dealii-adapter/blob/005933d6b45f885a64aee7ce597a3d7d81d54aff/.github/workflows/indent.yml#L22-L24) works here as well.

It prints the ill lines as well in the log.

```shell
--- original/./contrib/mapping-tester/preparemeshes.py
+++ fixed/./contrib/mapping-tester/preparemeshes.py
@@ -40,7 +40,7 @@
     data_name = "{}".format(function)
     [pathName, tmpfilename] = os.path.split(os.path.normpath(mainMesh))
     subprocess.run(["vtk_calculator.py", "--mesh", os.path.expandvars(file), "--function",
-                   function, "--data", data_name, "--directory", pathName, "-o", tmpfilename])
+                    function, "--data", data_name, "--directory", pathName, "-o", tmpfilename])
 
 
 def preparePartMesh(meshdir, name, p, force=False):
@@ -[6](https://github.com/precice/aste/pull/91/checks#step:4:6)2,[7](https://github.com/precice/aste/pull/91/checks#step:4:7) +62,7 @@
     os.makedirs(partDir, exist_ok=True)
     [pathName, tmpfilename] = os.path.split(os.path.normpath(partMesh))
     subprocess.run(["partition_mesh.py", "--mesh", mainMesh, "--algorithm",
-                   "meshfree", "-o", partMesh, "--directory", pathName, "-n", str(p)])
+                    "meshfree", "-o", partMesh, "--directory", pathName, "-n", str(p)])
```

DavidSCN commented:
Perfect, thanks! Then it should be easy to fix.

DavidSCN commented:
Could you rebase and update the PR description?
